src/app/theme/SplashScreenService.ts

```diff
@@ -180,45 +183,46 @@ const destroyTimer = (
   }
 }
 
-let activeTimersAtBackground: Record<string, boolean> = {}
+let activeTimersAtBackground: SplashScreenEnum[] = []
 
 const resetTimersOnActive = (): void => {
   splashScreenLogger.debug(
     `App is becoming active. Evaluating timers to reset...`
   )
 
-  const timersToReset = Object.keys(activeTimersAtBackground)
-
-  if (timersToReset.length === 0) {
+  if (activeTimersAtBackground.length === 0) {
     splashScreenLogger.info(`No splash screen timers to reset.`)
   } else {
-    timersToReset.forEach(splashScreen => {
-      if (activeTimersAtBackground[splashScreen]) {
-        splashScreenLogger.info(
-          `Resetting timer for splash screen "${splashScreen}"`
-        )
-        setTimeoutForSplashScreen(splashScreen as SplashScreenEnum)
-      }
+    activeTimersAtBackground.forEach(splashScreen => {
+      splashScreenLogger.info(
+        `Resetting timer for splash screen "${splashScreen}"`
+      )
+      setTimeoutForSplashScreen(splashScreen)
     })
   }
 
   // Clear the record once timers are reset
-  activeTimersAtBackground = {}
+  activeTimersAtBackground = []
 }
 
 const clearTimersOnBackground = (): void => {
   splashScreenLogger.debug(
     `App is going to background. Clearing active timers...`
   )
 
-  Object.keys(autoHideTimers).forEach(splashScreen => {
-    if (autoHideTimers[splashScreen]) {
+  const activeTimers = Object.keys(autoHideTimers) as SplashScreenEnum[]
+
+  activeTimers.forEach(splashScreen => {
+    const timerId = autoHideTimers[splashScreen]
+
+    if (timerId) {
```

Comment:
technically will always be defined but it's simpler in terms of typing to assume the impossible